### ISSUE TITLE
Improve rendering of close objects / POIs to improve real life usability

### ISSUE THREAD

**Issue by sid-logic** [osmandapp/OsmAnd#6177]
Hello everyone,

at first: Kudos for the great app. I haven't found anything in an already existing issue. If I missed it, sry.

## TL;DR

Please add some indication for POI's that will not be shown because something else would overlap with it. Good example is the OSM Mapnik style, which shows purple dots in that case.

## The problem 

I use Osmand as my main OSM app on a daily basis. But there is always one thing, that really bothers me: POI's that are **close together** may **not be visible** on the map. Let me quickly give you an example (comparison of Osmand offline vector render vs. online OSM Mapnik tiles).

![osmand_mapnik_compare_lowzoom](https://user-images.githubusercontent.com/5895949/46969295-e423a780-d0b5-11e8-8e94-7eb8ea11cf9a.png)
![osmand_mapnik_compare_highzoom](https://user-images.githubusercontent.com/5895949/46969572-d6baed00-d0b6-11e8-80d7-3cac9f2c2dd8.png)

Here is the location shown in the images: https://osm.org/go/0Gcf~_pVt

As you can see, in the low zoom (first comparison) there are quite a lot of POI's missing in Osmand. Mapnik doesn't show the POI's in detail, but it contains these small purple dots to indicate, that something is there, but you have to zoom in. 

The second image is even worse. Now I have a quite decent zoom level for walking usage (overview of the building). I've marked the missing POI's in the OSM online tile image. **You will miss 7 out of 17 POI's in total.** This is really bad!

## Solution

Osmand would really benefit something similar to the Mapnik rendering: Please show some **indicator for the discarded POI's**. Maybe you can keep the current logic to determine, which POI's should be visible and which should not. And then show the "not visible" POI's as small dots for example.

If you don't, then **many users** of this app will **miss out some cool stuff** just because they haven't zoomed in to the highest zoom level in order to grasp every POI that's available. 

## Further thinking

I've read something about [custom styles](https://osmand.net/help/docs/Custom_Rendering_How-To.htm) and maybe this can be used?

## Affected version

Android App (Stable @ Play Store) 
Osmand+ 3.2.2 (with Live Subscription)
More details is activated

**Comment by scaidermern** [comment]
Simplifying a POI icon to a small dot of the same color if it doesn't fit otherwise is a good idea.

**Comment by quax2712** [comment]
If you change screen zoom to 50% you will see all POI's. (long press '+' button on screen for zoom menu).
Else you must change default.render.xml file and save it to ..\Android\data\net.osmand.plus\files\rendering\ or create a new file and make it depend on default.

**Comment by siduenho** [comment]
@quax2712 Thanks for the tip. The zoom setting with 50% seems to help for the most part. But this seems more to be a workaround than a "standard user" solution.

I the casual user won't change the zoom settings, thus the problem still persists.

I've already looked into the rendering xml but I'm not sure, where and how this "overlapping nodes will be hidden" feature is configured.

**Comment by vshcherb** [comment]
We have that solution for POI search but not for rendering map style

**Comment by siduenho** [comment]
Just a sidenote: This feature is somewhat available for OSM notes, too.

**Comment by mikehgentry** [comment]
Another option would be to draw some of the hidden ones underneath but slightly offset, so they look kind of like a stack of poker chips viewed from almost directly above (up to some sensible limit - 3 or 5 or something, beyond that you only need to know that there's 'lots')
